Sources/Datadog/DDTracer.swift

```diff
@@ -103,6 +103,17 @@ public class DDTracer: Tracer {
 
     // MARK: - Internal
 
+    internal func createSpanContext() throws -> DDSpanContext {
```

Comment:
Let's use this method to create the context in `startSpanOrThrow()`, so there's only one control flow for the `DDSpanContext` creation. Otherwise, we'll have to test it separately.

Also, it's better for this method to accept `tracing: TracingFeature` as an argument.
* this will centralize the error throwing (less testing - now we'd have to test as well if this method really throws);
* this will clearly mean that the `URLSession` auto-instrumentation depends on the `TracingFeature.instance`.